src/arrayMethodJoin.js

```diff
@@ -4,8 +4,30 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const currentArrayLength = this.length;
+    const joinSeparator = String(separator);
+    let resultingString = '';
+
+    if (!currentArrayLength) {
+      return resultingString;
+    }
```

Comment:
Do we really need this?
If length === 0 we will skip `forEach` and also return `resultingString`